doc/languages-frameworks/lisp.section.md

```diff
@@ -0,0 +1,321 @@
+compilation.
+
+The `build-asdf-system` function is documented with comments in
+`nix-cl.nix`. Also, `packages.nix` is full of examples of how to use it.
+
+## Defining packages manually outside Nixpkgs
+
+Lisp derivations (`abcl`, `sbcl` etc.) also export the `buildASDFSystem`
+function, which is the same as `build-asdf-system`, except for the `lisp`
+argument which is set to the given CL implementation.
+
+It can be used to define packages outside Nixpkgs, and, for example, add them
+into the package scope with `override` and `packageOverrides`, both of which
+will be discussed later on.
+
+### Including an external package in scope
+
+A package defined outside Nixpkgs using `buildASDFSystem` can be woven into the
+Nixpkgs-provided scope like this:
+
+```
+let
+  alexandria = sbcl.buildASDFSystem rec {
+    pname = "alexandria";
+    version = "1.4";
+    src = fetchFromGitLab {
+      domain = "gitlab.common-lisp.net";
+      owner = "alexandria";
+      repo = "alexandria";
+      rev = "v${version}";
+      hash = "sha256-1Hzxt65dZvgOFIljjjlSGgKYkj+YBLwJCACi5DZsKmQ=";
+    };
+  };
+  sbcl' = sbcl.override {
+    packageOverrides = self: super: {
+      inherit alexandria;
+    };
+  };
+in sbcl'.pkgs.alexandria
+```
+
+## Overriding package attributes
+
+Packages export the `overrideLispAttrs` function, which can be used to build a
+new package with different parameters.
+
+Example of overriding `alexandria`:
+
+```
+sbcl.pkgs.alexandria.overrideLispAttrs (oldAttrs: rec {
+  version = "1.4";
+  src = fetchFromGitLab {
+    domain = "gitlab.common-lisp.net";
+    owner = "alexandria";
+    repo = "alexandria";
+    rev = "v${version}";
+    hash = "sha256-1Hzxt65dZvgOFIljjjlSGgKYkj+YBLwJCACi5DZsKmQ=";
+  };
+})
+```
+
+## Overriding packages in scope
+
+Packages can be woven into a new scope by using `override` with
+`packageOverrides`:
+
+```
+let
+  sbcl' = sbcl.override {
+    packageOverrides = self: super: {
+      alexandria = super.alexandria.overrideLispAttrs (oldAttrs: rec {
+        pname = "alexandria";
+        version = "1.4";
+        src = fetchFromGitLab {
+          domain = "gitlab.common-lisp.net";
+          owner = "alexandria";
+          repo = "alexandria";
+          rev = "v${version}";
+          hash = "sha256-1Hzxt65dZvgOFIljjjlSGgKYkj+YBLwJCACi5DZsKmQ=";
+        };
+      });
+    };
+  };
+in builtins.elemAt sbcl'.pkgs.bordeaux-threads.lispLibs 0
+```
+
+### Dealing with slashy systems
+
+Slashy (secondary) systems should not exist in their own packages! Instead, they
+should be included in the parent package as an extra entry in the `systems`
+argument to the `build-asdf-system`/`buildASDFSystem` functions.
+
+The reason is that ASDF searches for a secondary system in the `.asd` of the
+parent package. Thus, having them separate would cause either one of them not to
+load cleanly, because one will contains FASLs of itself but not the other, and
+vice versa.
```

Comment:
Of course, there are some bad cases like `cl-ppcre/unicode` → `cl-unicode` → `cl-ppcre` (if I remember correctly)… I think even if there is nothing encouraging to tell about such situations, maybe it is better to mention as something to look out for?

Reply:
Are you thinking about circular dependencies?

What would you add to the manual?

Reply:
Well, if we have no generic advice to offer, we could add after the example 
«
Note that sometimes the slashy systems might not only have more dependencies than the main one, but create a circular dependency between `.asd` files. Unfortunately, in this case an adhoc solution becomes necessary.
»

Reply:
Added https://github.com/NixOS/nixpkgs/compare/457303158f1fb67dded7b6d89f75a122c911ac14..2f70da2da9d57291024e2989b9016f5f49924fb6